### ISSUE TITLE
Password Options

### ISSUE THREAD

**Issue by andy61i** [FifiTheBulldog/ios-settings-urls#73]
Hi All! 

I am looking for an URL to open Settings > Passwords > **Password Options**
Some password managers like 1password & LastPass can navigate there. 
What I found in public sources is only App-prefs:PASSWORDS that opens only Settings > Passwords

**Comment by mihirwagle** [comment]
The following should work for OS Version < 18.0
prefs:root=PASSWORDS&path=PASSWORD_OPTIONS

**Comment by andy61i** [comment]
> The following should work for OS Version < 18.0 prefs:root=PASSWORDS&path=PASSWORD_OPTIONS

But it doesn't

**Comment by keepassium** [comment]
We have it working on iOS 17.4 with `App-prefs:PASSWORDS&path=PASSWORD_OPTIONS`

**Comment by mihirwagle** [comment]
> > The following should work for OS Version < 18.0 prefs:root=PASSWORDS&path=PASSWORD_OPTIONS
> 
> But it doesn't

I was using this successfully in 17.6.1 but it's broken in 18. Curious why it is broken for you in pre 18.0

**Comment by andy61i** [comment]
> App-prefs:PASSWORDS&path=PASSWORD_OPTIONS

You are right. Retested and it does work on iOS 17. Thought on iOS 16 it only enters Passwords, not Password Options.

**Comment by Sanbolee** [comment]
> > App-prefs:PASSWORDS&path=PASSWORD_OPTIONS
> 
> You are right. Retested and it does work on iOS 17. Thought on iOS 16 it only enters Passwords, not Password Options.

Have you found the code on iOS16

**Comment by llvllo** [comment]
> Hi All!
> 
> I am looking for an URL to open Settings > Passwords > **Password Options** Some password managers like 1password & LastPass can navigate there. What I found in public sources is only App-prefs:PASSWORDS that opens only Settings > Passwords
> 
> UPDATE: And also looking for a iOS 18 variant of this to open to open Setting > General > AutoFill & Passwords

URL for AutoFill & Passwords: settings-navigation://com.apple.Settings.General/AUTOFILL
Tested on iPadOS 18.4 and iOS 18.5 Beta 2

**Comment by andy61i** [comment]
> URL for AutoFill & Passwords: settings-navigation://com.apple.Settings.General/AUTOFILL Tested on 18.5 Beta 2

Hi llvllo! Thanks man! This URL is not working for 18.4.1, but we'll be testing it on 18.5 later on. Thanks again!

**Comment by llvllo** [comment]
> > URL for AutoFill & Passwords: settings-navigation://com.apple.Settings.General/AUTOFILL Tested on 18.5 Beta 2
> 
> Hi llvllo! Thanks man! This URL is not working for 18.4.1, but we'll be testing it on 18.5 later on. Thanks again!

I just tried it on my iPad on 18.4 and can confirm it is working, I'll update my iPad and re check it! Also since iOS 18, Passwords were moved to a new App, you can use its bundle ID to open that and any other app. The bundle ID for passwords app is: com.apple.Passwords

You can use this list to find bundle IDs for a bunch of Apple's apps: https://support.apple.com/guide/deployment/bundle-ids-for-native-iphone-and-ipad-apps-depece748c41/web

Update: Tested and confirmed to be working on iPadOS 18.4.1

I used a URL and then linked it to Open with the URL type as text.

**Comment by andy61i** [comment]
> I used a URL and then linked it to Open with the URL type as text.

Probably this is why. Opening the link programmatically gives an error, but strangely works if select the link as a text (in any note taking app) and call Open Link from the context menu.

**Comment by llvllo** [comment]
> > I used a URL and then linked it to Open with the URL type as text.
> 
> Probably this is why. Opening the link programmatically gives an error, but strangely works if select the link as a text (in any note taking app) and call Open Link from the context menu.

Yeah not sure as to why this is, I found a document with a large list of links and they all seem to work on later versions. If you use URL and then use Open and link the 2, after they are linked click on the URL in Open and define you want it to be text it will open from shortcuts and work. I can provide a sample if you would like.

**Comment by llvllo** [comment]
![Image](https://github.com/user-attachments/assets/ceb3e02c-616d-43b6-b538-7c7e629f93be)

**Comment by llvllo** [comment]
Updated AutoFill & Passwords URL: prefs:root=General&path=AUTOFILL
Tested on iPadOS 18.4.1 & iOS 18.5 Beta 2

EDIT: no need to change this to Text it will work as a URL

**Comment by andy61i** [comment]
I do confirm that prefs:root=General&path=AUTOFILL and settings-navigation://com.apple.Settings.General/AUTOFILL work from Shortcuts app on iOS 18.4. Unfortunately, none of them works programmatically like that

`UIApplication.shared.open(url, options: [:], completionHandler: nil)`
